doc/src/contributing/new-contributors-guide/dev-setup.md

```diff
@@ -0,0 +1,228 @@
+
+```{note}
+This guide is intended for people have never contributed to an open source
+project on GitHub before. If you are already familiar with how to contribute
+to an open source project on GitHub, go to the [](./workflow-process.md) guide
+```
+
+The first step to contributing to the code base is creating your development environment.
+
+```{important}
+Each of the steps in this guide only need to be done once. Once you have
+completed them, you do not need to repeat them, even if you are making a
+second contribution.
+```
+
+## Install Git
+
+SymPy is available on [GitHub](https://github.com/sympy/sympy) and uses
+[Git](https://git-scm.com) for source control. The workflow is such that
+code is pulled and pushed to and from the main repository. Install the respective version
+of Git for your operating system to start development.
+
+**Linux-like systems**:
+
+Install git via your native package management system:
+
+```
+yum install git
+```
+
+or:
+
+```
+sudo apt-get install git
+```
+
+**Windows and macOS**:
+
+The easiest way to get git is to download [GitHub
+desktop](https://desktop.github.com/), which will install git, and also
+provide a nice GUI (this tutorial will be based on the command line
+interface). Note, you may need to go into the GitHub preferences and choose
+the "Install Command Line Tools" option to get git installed into the
+terminal.
+
+If you do decide to use the GitHub GUI, you should make sure that any "sync
+does rebase" option is disabled in the settings.
+
+### Configure Your Name and Email in Git
+
+Git tracks who makes each commit by checking the user’s name and email.
+In addition, we use this info to associate your commits with your GitHub account.
+
+To set these, enter the code below, replacing the name and email with your own (`--global` is optional).:
+
+```
+git config --global user.name "Firstname Lastname"
+git config --global user.email "your_email@youremail.com"
+```
+
+The name should be your actual name, not your GitHub username. Use the email you used for your GitHub account (see [below](dev-setup-create-github-account)).
+
+### (Optional) Configure Git Settings
+
+*This step is not required, but it can make working with git on the command
+line easier.*
+
+These global options (i.e. applying to all repositories) are placed in
+`~/.gitconfig`. If you want, you can edit this file to enable some handy
+shortcuts:
+
+```
+[user]
+    name = Firstname Lastname
+    email = your_email@youremail.com
+
+# Some helpful aliases to save on typing
+[alias]
+    ci = commit
+    di = diff --color-words
+    st = status
+    co = checkout
+    log1 = log --pretty=oneline --abbrev-commit
+    logs = log --stat
+
+```
+
+See <https://git-scm.com/book/sv/v2/Customizing-Git-Git-Configuration> for
+some more common git configuration options.
+
+### (Optional) Tune Your Bash Prompt for Git
+
+*This step is not required, but it can make working with git on the command
+line easier, so it is recommended.*
```

Comment:
I would move or remove this whole section. It is a distraction compared to the important things like "open a github account" which are further down.

Reply:
I removed it. It is definitely something I recommend to new git users, but I'm also starting to lean towards we should have little to no very intro git stuff in our dev guide. So including on the very basics is a decent compromise for now.

Reply:
I think that this is useful information but it would just be better on a separate "git tips" page. Right here there shouldn't be anything more than a link to that page though because none of this is really needed and it's important not to overwhelm people with unnecessary things.